tools/config-generator/getlivebranch.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/google/go-github/v27/github"
+	"knative.dev/pkg/test/ghutil"
+)
+
+func latestReleaseBranch(gc *ghutil.GithubClient, repo string) (string, error) {
+	parts := strings.Split(repo, "/")
+	if len(parts) != 2 {
+		return "", fmt.Errorf("repo name %q should be in the form of [ORG]/[REPO]", repo)
+	}
+	branches, err := gc.ListBranches(parts[0], parts[1])
+	if err != nil {
+		return "", fmt.Errorf("failed listing branches for repo %q: %w", repo, err)
+	}
+	return latestReleaseBranches(branches), nil
+}
+
+func latestReleaseBranches(branches []*github.Branch) string {
+	var (
+		reReleaseBranch = regexp.MustCompile(`^release\-(\d+\.\d+)$`)
+		latest          = "0.0"
```

Comment:
Is "0.0" a valid return value for this function or should we capture a failure?

Reply:
`release-0.0` is not a vaild release, and shouldn't appear at all. So `0.0` just means there is no release branch, I think I can add a comment to make it clearer

Reply:
Updated

Reply:
I'm more concerned about a scenario where a `release-*` branch is not localized and "0.0" is returned. This will hide effectively the error when it seems this would be the best place to manage it.

Reply:
I agree that this is the best place finding errors. But it's not super clear to me what scenario should trigger an error. i.e. when there is a branch named `release-something` it might be intended, and shouldn't fail this tool

Reply:
What will happen if `latestReleaseBranch` returns `release-0.0` as the latest release branch? If the callsite will consider it as an error and skip it, I think it makes more sense to return an error here rather than returning a relatively weird non-existent branch name.

Reply:
Makes sense. Changed to make the function return "" instead of "0.0", and add a condition at caller for this check